### ISSUE TITLE
Router is unreliable

### ISSUE THREAD

**Issue by paul-butcher** [wellcomecollection/platform#5463]
This time yesterday, the router dlq contained about 50 items.

Right now, it contains over 300.

**Comment by paul-butcher** [comment]
It may also mean that the relation embedder is running in a suboptimal fashion.

If the embedder processes each message individually, it could be running several times for each message into the router.

**Comment by paul-butcher** [comment]
This is blocked by https://github.com/wellcomecollection/platform/issues/5471

**Comment by paul-butcher** [comment]
One notable thing here is that router fetches the entire set of documents from the database, then iterates over them, doing the following:

If it has a collectionPath, put the collectionPath on the relation_embedder queue.
If it does not, transition the document to Denormalised.

In the "has collectionPath" case, this seems rather wasteful.  I wonder if this can be improved by only asking for what is required:
e.g.
```
GET /works-merged-2022-03-18/_mget
{
  "docs": [
    {
      "_id": "dcfbb52f",
      "_source":["data.collectionPath"]
    },
    {
      "_id": "h7tdnm89",
      "_source":["data.collectionPath"]
    }
  ]
}
```

I also wonder if it might be more efficient to explicitly use the index-in-path form of _mget

**Comment by paul-butcher** [comment]
I think there's a question here about the way things progress through the pipeline by being fetched then inserted into the next database.  In the case of the whole relation embedder, the body of the Work is unchanged, but the whole document still gets pulled and pushed.

It may be better to only fetch collectionPath and state, and only save state.

This suggests that state and the work it refers to may be distinct objects to be saved in different indices.  The Work can then just stay where it is across this particular stage.

**Comment by paul-butcher** [comment]
Some original discussion here: https://github.com/wellcomecollection/docs/pull/42

**Comment by paul-butcher** [comment]
WRT batchRetrieveFlow, it just occurred to me that one likely reason for spikes downstream is its use upstream.  So where I say that router very occasionally sees some spikes, it's probably partly because matcher or id_minter has been holding them back before opening the sluice.

**Comment by paul-butcher** [comment]
The change I made to fix this initially is now in production.  A better fix is coming in the form of #5475
